setup.py

```diff
@@ -1,49 +1,34 @@
-#!/usr/bin/env python
-
+from configparser import ConfigParser
 import os
-from setuptools import setup, find_packages
+from setuptools import setup
+from pathlib import Path as _Path
+
+
+def setup_package():
+    write_requirements()
+    setup()
+
 
-version_file = os.path.join(os.path.dirname(__file__), "dirty_cat", "VERSION.txt")
-with open(version_file) as fh:
-    VERSION = fh.read().strip()
+def write_requirements():
+    """Parse dependencies from setup.cfg and write them
+    within the project scope.
 
-description_file = os.path.join(os.path.dirname(__file__), "README.rst")
-with open(description_file) as fh:
-    DESCRIPTION = fh.read()
+    Notes
+    -----
+    It will allow to compare installed versions against
+    required versions at run time, during the import of this package.
+    """
+    setup_file = "setup.cfg"
+    config = ConfigParser()
+    config.read(setup_file)
+    deps = config["options"]["install_requires"].strip()
+    project_name = config["metadata"]["name"]
+    project_dir = _Path(__file__).parent / project_name
+    req_path = os.path.join(project_dir, "_config_requirements.py")
+    with open(req_path, "w+") as f:
+        body = "deps = " + str({"deps": deps})
+        f.write(body)
```

Comment:
As we're using pathlib, I guess it could be
```suggestion
    req_path = project_dir / "_config_requirements.py"
    req_path.write_text(f"deps = { {"deps": deps} }")
```
(not 100% sure this will run as expected)
Also, is there a specific reason you used "w+"? As you're only writing to the file, I assumed you would only need "w".

Reply:
Yes, we should use `pathlib` here. And I think I put `"w+"` by pure reflex but yes we can limit it to `"w"`.